codegen/templates/module_initializer.tmpl

```diff
@@ -35,21 +33,25 @@ type {{$className | title}}DependenciesNodes struct {
 // InitializeDependencies fully initializes all dependencies in the dep tree
 // for the {{$instance.InstanceName}} {{$instance.ClassName}}
 func InitializeDependencies(
-	gateway *zanzibar.Gateway,
+	g *zanzibar.Gateway,
 ) (*DependenciesTree, *Dependencies) {
-	{{- if not $instance.HasDependencies}}
-	return nil, {{$instance.PackageInfo.ExportName}}(gateway)
-	{{- else}}
 	tree := &DependenciesTree{}
 
-	{{- range $idx, $className := $instance.DependencyOrder}}
+	initializedDefaultDependencies := &zanzibar.DefaultDependencies{
+		Logger:  g.Logger,
+		Scope:   g.AllHostScope,
+		Config:  g.Config,
+		Channel: g.Channel,
```

Comment:
What's the use case for the `Channel` here ?

Reply:
Ok I understand now. Allocating sub channels and creating custom tchannel clients.

Reply:
Yes, it's needed for tchannel clients. I was trying to see if we could customize the default dependencies, so only `Channel` is added for tchannel clients, but it became complicated.